[forecast] Add an explanation of the format of a forecast query string (#8183)
DataDog/documentation: content/en/monitors/monitor_types/forecasts.md

## API

To create forecast monitors programmatically, see the [Datadog API reference][7]. Datadog **strongly recommends** [exporting a monitor's JSON][8] to build the query for the API. By using the [monitor creation page][1] in Datadog, customers benefit from the preview graph and automatic parameter tuning to help avoid a poorly configured monitor.

Forecast monitors are managed using the [same API][9] as other monitors, but the contents of the `query` property deserves further explanation.

The `query` property in the request body should contain a query string in the following format:

```text
<aggregator>(<query_window>):forecast(<metric_query>, '<algorithm>', <deviations>, interval=<interval>[, history='<history>'][, model='<model>'][, seasonality='<seasonality>']) <comparator> <threshold>
```

* `aggregator`: `min` if the alert should trigger when the forecast goes below the threshold, and `max` if the alert should trigger when the forecast goes above the threshold 

> * `query_window`: a timeframe like `last_4h` or `last_7d`; the time window displayed in graphs in notifications; must be at least as large as the `alert_window` and is recommended to be around 5 times the `alert_window`

Reviewer: ```suggestion
* `query_window`: a timeframe, for example: `last_4h` or `last_7d`. It is recommended that the time window displayed in graphs and notifications be five times the `alert_window`, but must be at least the size of the `alert_window`.
```

> * `metric_query`: a standard Datadog metric query (e.g., `min:system.disk.free{service:database,device:/data}by{host}`)

Reviewer: ```suggestion
* `metric_query`: a standard Datadog metric query, for example: `min:system.disk.free{service:database,device:/data}by{host}`
```

* `algorithm`: `linear` or `seasonal`

> * `deviations`: a number greater than or equal to 1; controls the size of the confidence bounds, allowing a monitor to be made more or less sensitive

Reviewer 2: should we keep the same definition for deviations for both anomalies and forecasts for consistency?

Author: The range of expected values is actually a little different. For anomalies, a bounds arg of 0.5 is fine, but for forecast it doesn't really make sense. I used the "confidence bounds" terminology here because this is how we describe it in the forecast monitor UI (which is different than the description in the anomalies UI).

> * `deviations`: a number greater than or equal to 1; controls the size of the confidence bounds, allowing a monitor to be made more or less sensitive

Reviewer: ```suggestion
* `deviations`: a number greater than or equal to one, which controls the size of the confidence bounds, allowing a monitor to be made more or less sensitive.
```

* `interval`: a positive integer representing the number of seconds in the rollup interval

> * `history`: a string representing the amount past data that should be used when making the forecast (e.g., `1w`, `3d`); this parameter is only for use with the `linear` algorithm

Reviewer: ```suggestion
* `history`: a string representing the amount of past data that should be used for making the forecast, for example: `1w`, `3d`. This parameter is only used with the `linear` algorithm.
```

> * `model`: `default`, `simple`, or `reactive`; this parameter is only for use with the `linear` algorithm

Reviewer: ```suggestion
* `model`: The type of model to use: `default`, `simple`, or `reactive`. This parameter is only used with the `linear` algorithm.
```

> * `seasonality`: `hourly`, `daily`, or `weekly`; this parameter is only for use with the `seasonal` algorithm

Reviewer: ```suggestion
* `seasonality`: The seasonality to use: `hourly`, `daily`, or `weekly`. This parameter is only used with the `seasonal` algorithm
```

> * `comparator`: `<=` to alert when the forecast goes below the threshold; `>=` to alert when the forecast goes above the threshold

Reviewer 2: It feels a bit redundant with the `aggregator` param. Can the user send a `min` aggregator with a `>=` (or do we block it), if so should we specify it ?

Author: You're right; they need to be synchronized. Good question about validation on the backend... I'll have to check if we actually enforce it, but if a monitor was created with these two things mismatched, it'd act very strangely...

Author: It doesn't look like we explicitly block conflicting aggregators and comparators, and the monitor should evaluate, but it would give undesirable results.

Reviewer 2: Do you think we should create a card to prevent this behavior ?

Author: 👍 validation for this has been implemented and will be rolled out shortly.

> * `comparator`: `<=` to alert when the forecast goes below the threshold; `>=` to alert when the forecast goes above the threshold

Reviewer: ```suggestion
* `comparator`: Use `<=` to alert when the forecast goes below the threshold. Use `>=` to alert when the forecast goes above the threshold.
```

> * `threshold`: a number; a critical alert will trigger when the confidence bounds around the forecast reach this threshold

Reviewer: ```suggestion
* `threshold`: A number used to trigger a critical alert when the confidence bounds around the forecast reach the threshold.
```
